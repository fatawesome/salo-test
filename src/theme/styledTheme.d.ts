import 'styled-components';

declare module 'styled-components' {
  export interface DefaultTheme {
    boxShadow: string;
    colors: {
      background: '#FFFFFF';

      cta: string; // buttons, checkboxes, price (i.e. everything screaming on user)
      textPrimary: string;
      textSecondary: string;
    };
  }
}
