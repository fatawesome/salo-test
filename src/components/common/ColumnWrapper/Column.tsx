import styled from 'styled-components';

const ColumnWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export default ColumnWrapper;
