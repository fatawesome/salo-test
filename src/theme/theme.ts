import { DefaultTheme } from 'styled-components';

// TODO: font and stuff;
const aviasalesTheme: DefaultTheme = {
  boxShadow: '0px 2px 8px rgba(0, 0, 0, 0.1)',
  colors: {
    textPrimary: '#4A4A4A',
    textSecondary: '#A0B0B9',
    background: '#FFFFFF',
    cta: '#2196F3',
  }
}

export { aviasalesTheme };
