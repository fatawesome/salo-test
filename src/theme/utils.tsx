import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { aviasalesTheme } from './theme';
import { render } from '@testing-library/react';

const withTheme = (node: React.ReactNode) => (
  <ThemeProvider theme={aviasalesTheme}>
    {node}
  </ThemeProvider>
)

export const snapshotWithTheme = (node: React.ReactNode) => {
  return renderer.create(withTheme(node));
}

export const uiWithTheme = (node: React.ReactNode) => {
  return render(withTheme(node));
}

// TODO: можно обобщить в функцию вида (fn :: renderer -> node -> result)
