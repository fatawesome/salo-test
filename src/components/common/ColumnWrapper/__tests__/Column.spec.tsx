import React from 'react';

import Column from '../Column';
import { snapshotWithTheme } from '../../../../theme/utils';

describe('Column component', () => {
  test('renders', () => {
    const component = snapshotWithTheme(
      <Column>
        <div>Row 1</div>
        <div>Row 2</div>
        <div>Row 3</div>
      </Column>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
