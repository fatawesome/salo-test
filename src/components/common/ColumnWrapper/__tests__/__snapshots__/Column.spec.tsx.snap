// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component renders 1`] = `
<div
  className="Column__ColumnWrapper-sc-gvqm9n-0 cIEksv"
>
  <div>
    Row 1
  </div>
  <div>
    Row 2
  </div>
  <div>
    Row 3
  </div>
</div>
`;
